superbench/benchmarks/micro_benchmarks/directx_gemm_flops_performance/GPUCore.cpp

```diff
@@ -0,0 +1,507 @@
+    bindingDescB.TotalTensorSizeInBytes = DMLCalcBufferTensorSize(dataType, tensorElementCount);
+
+    dmlTensorDescB.Desc = &bindingDescB;
+    dmlGEMMOperatorDesc.BTensor = &dmlTensorDescB;
+
+    DML_TENSOR_DESC dmlTensorDescC{};
+    dmlTensorDescC.Type = DML_TENSOR_TYPE_BUFFER;
+
+    DML_BUFFER_TENSOR_DESC bindingDescC = {};
+    UINT tensorSizes[4] = {1, 1, static_cast<UINT>(m), static_cast<UINT>(n)};
+    tensorElementCount = tensorSizes[0] * tensorSizes[1] * tensorSizes[2] * tensorSizes[3];
+    bindingDescC.DataType = dataType;
+    bindingDescC.Flags = DML_TENSOR_FLAG_NONE;
+    bindingDescC.DimensionCount = ARRAYSIZE(tensorSizes);
+    bindingDescC.Sizes = tensorSizes;
+    bindingDescC.Strides = nullptr;
+    bindingDescC.TotalTensorSizeInBytes = DMLCalcBufferTensorSize(dataType, tensorElementCount);
+
+    dmlTensorDescC.Desc = &bindingDescC;
+
+    dmlGEMMOperatorDesc.OutputTensor = &dmlTensorDescC;
+    dmlGEMMOperatorDesc.CTensor = nullptr;
+    dmlGEMMOperatorDesc.TransA = DML_MATRIX_TRANSFORM_NONE;
+    dmlGEMMOperatorDesc.TransB = DML_MATRIX_TRANSFORM_NONE;
+    dmlGEMMOperatorDesc.Alpha = 1.0f;
+    dmlGEMMOperatorDesc.Beta = 0.0f;
+
+    DML_OPERATOR_DESC dmlOperatorDesc{DML_OPERATOR_GEMM, &dmlGEMMOperatorDesc};
+
+    ThrowIfFailed(m_dmlDevice->CreateOperator(&dmlOperatorDesc, IID_PPV_ARGS(dmlOperator.GetAddressOf())));
+
+    ThrowIfFailed(m_dmlDevice->CompileOperator(dmlOperator.Get(), DML_EXECUTION_FLAG_NONE,
+                                               IID_PPV_ARGS(m_dmlCompiledOperator.GetAddressOf())));
+}
+
+/**
+ * @brief Prepare input and output data and buffers of the tensor elements..
+ */
+template <typename T> void GPUCore::PrepareData(const int m, const int n, const int k) {
+    // Define the tensors.
+    std::vector<T> dataA(m * k);
+    std::vector<T> dataB(n * k);
+
+    // Prepare input data.
+    std::fill(dataA.begin(), dataA.end(), 1);
+    std::fill(dataB.begin(), dataB.end(), 1);
+
+    UINT64 byteSize = m * k * sizeof(T);
+
+    // Setup input buffer A and upload input data.
+    m_inputBufferA =
+        CreateDefaultBuffer(m_device.Get(), m_commandList.Get(), dataA.data(), byteSize, m_inputUploadBufferA);
+
+    byteSize = n * k * sizeof(T);
+    // Setup input buffer B and upload input data.
+    m_inputBufferB =
+        CreateDefaultBuffer(m_device.Get(), m_commandList.Get(), dataB.data(), byteSize, m_inputUploadBufferB);
+
+    byteSize = m * n * sizeof(T);
+    // Create output buffer.
+    m_device->CreateCommittedResource(
+        get_rvalue_ptr(CD3DX12_HEAP_PROPERTIES(D3D12_HEAP_TYPE_DEFAULT)), D3D12_HEAP_FLAG_NONE,
+        get_rvalue_ptr(CD3DX12_RESOURCE_DESC::Buffer(byteSize, D3D12_RESOURCE_FLAG_ALLOW_UNORDERED_ACCESS)),
+        D3D12_RESOURCE_STATE_UNORDERED_ACCESS, nullptr, IID_PPV_ARGS(&m_outputBuffer));
+
+    // Create readback buffer.
+    m_device->CreateCommittedResource(get_rvalue_ptr(CD3DX12_HEAP_PROPERTIES(D3D12_HEAP_TYPE_READBACK)),
```

Comment:
>CreateCommittedResource [](http://example.com/codeflow?start=14&length=23)

I suggest we need check the return value of the D3D functions to see if it works, like this function may return E_OUTOFMEMORY when resource cannot be allocated

Reply:
add throwiffailed to all CreateCommittedResource